process/factory/indexerFactory.go

```diff
@@ -53,6 +57,15 @@ func NewIndexer(args ArgsIndexerFactory) (dataindexer.Indexer, error) {
 		return nil, err
 	}
 
+	if args.SovereignConfig {
```

Comment:
No if/else here, please.
RunTypeComp creation should be at the highest level (main file) and passed down.

Reply:
I had to create runType components here because here is where the node is connecting with indexer factory